Sources/Components/NativeAdvert/NativeContentAdvertView.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import UIKit
+
+public final class NativeContentAdvertView: UIView {
+
+    // MARK: - Public properties
+
+    public weak var delegate: NativeAdvertViewDelegate?
+    public weak var imageDelegate: NativeAdvertImageDelegate?
+
+    // MARK: - Private properties
+
+    private let containerMaxWidth: CGFloat = 320
+    private let logoImageSize: CGFloat = 48.0
+    private let cornerRadius: CGFloat = 8.0
+
+    // See specification on:
+    // https://annonseweb.schibsted.no/nb-no/product/finn-native-ads-16031
+    private let imageAspectRatio: CGFloat = (1200.0 / 627)
+
+    // MARK: - UI properties
+
+    private lazy var containerView: UIView = {
+        let view = UIView(withAutoLayout: true)
+        view.layer.cornerRadius = cornerRadius
+        view.layer.masksToBounds = true
+        view.backgroundColor = UIColor(r: 248, g: 248, b: 248)!
+        return view
+    }()
+
+    private lazy var mainImageView: UIImageView = {
+        let view = UIImageView(withAutoLayout: true)
+        view.contentMode = .scaleAspectFill
+        view.clipsToBounds = true
+        return view
+    }()
+
+    private lazy var logoImageView: UIImageView = {
+        let view = UIImageView(withAutoLayout: true)
+        view.contentMode = .scaleAspectFit
+        return view
+    }()
+
+    private lazy var titleLabel: Label = {
+        let label = Label(style: .title2)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.numberOfLines = 0
+        label.minimumScaleFactor = 0.5
```

Comment:
I guess such a low scale-factor is is set `just in case the title is loooooooooooo…`?

Reply:
Yeah – or contains a lot of wide characters. We need to support 50 characters, period – I interpreted that to mean 50 *wide* characters, such as `M` as well 🙂